libs/webb-ui-components/src/hooks/useDarkMode.ts

```diff
@@ -22,78 +23,48 @@ export type ToggleThemeModeFunc = (
 export function useDarkMode(
   defaultTheme: SupportTheme = 'dark'
 ): [boolean, ToggleThemeModeFunc] {
-  const [preferredTheme, setPreferredTheme] =
-    useState<SupportTheme>(defaultTheme);
+  const [theme, setTheme] = useLocalStorageState('theme', {
+    defaultValue: defaultTheme,
+  });
 
-  const isDarkMode = useMemo(
-    () =>
-      isBrowser() && localStorage.getItem('theme') !== null
-        ? localStorage.getItem('theme') === 'dark'
-        : preferredTheme === 'dark',
-    [preferredTheme]
-  );
+  const isDarkMode = useMemo(() => theme === 'dark', [theme]);
 
   useEffect(() => {
     if (!isBrowser()) {
       return;
     }
 
-    if (localStorage.getItem('theme') === null) {
-      document.documentElement.classList.add('dark');
-      localStorage.setItem('theme', 'dark');
-      window.dispatchEvent(new Event('storage'));
-    }
-
-    if (
-      localStorage.getItem('theme') === 'dark' ||
-      (!('theme' in localStorage) &&
-        window.matchMedia('(prefers-color-scheme: dark)').matches)
```

Comment:
It would be best if you didn't remove the `prefers-color-scheme` check here, as it helps fetch the current OS theme.